tests/clients/test_urllib3.py

```diff
@@ -1,501 +1,530 @@
 # (c) Copyright IBM Corp. 2021
 # (c) Copyright Instana Inc. 2020
 
+import sys
+import unittest
 from multiprocessing.pool import ThreadPool
 from time import sleep
-import unittest
 
-import urllib3
+import pytest
 import requests
-
-import tests.apps.flask_app
-from ..helpers import testenv
+import urllib3
 from instana.singletons import agent, tracer
 
+import tests.apps.flask_app  # noqa: F401
+from tests.helpers import testenv
+
 
 class TestUrllib3(unittest.TestCase):
-    def setUp(self):
-        """ Clear all spans before a test run """
+    def setUp(self) -> None:
+        """Clear all spans before a test run"""
         self.http = urllib3.PoolManager()
-        self.recorder = tracer.recorder
+        self.recorder = tracer.span_processor
         self.recorder.clear_spans()
 
-    def tearDown(self):
-        """ Ensure that allow_exit_as_root has the default value """
+    def tearDown(self) -> None:
+        """Ensure that allow_exit_as_root has the default value"""
         agent.options.allow_exit_as_root = False
 
-    def test_vanilla_requests(self):
-        r = self.http.request('GET', testenv["wsgi_server"] + '/')
-        self.assertEqual(r.status, 200)
+    def test_vanilla_requests(self) -> None:
+        r = self.http.request("GET", testenv["wsgi_server"] + "/")
+        assert r.status == 200
 
         spans = self.recorder.queued_spans()
-        self.assertEqual(1, len(spans))
+        assert len(spans) == 1
 
-    def test_parallel_requests(self):
+    def test_parallel_requests(self) -> None:
         http_pool_5 = urllib3.PoolManager(num_pools=5)
 
         def task(num):
-           r = http_pool_5.request('GET', testenv["wsgi_server"] + '/', fields={'num': num})
-           return r
+            r = http_pool_5.request(
+                "GET", testenv["wsgi_server"] + "/", fields={"num": num}
+            )
+            return r
 
         with ThreadPool(processes=5) as executor:
             # iterate over results as they become available
             for result in executor.map(task, (1, 2, 3, 4, 5)):
-                self.assertEqual(result.status, 200)
+                assert result.status == 200
 
         spans = self.recorder.queued_spans()
-        self.assertEqual(5, len(spans))
-        nums = map(lambda s: s.data['http']['params'].split('=')[1], spans)
-        self.assertEqual(set(nums), set(('1', '2', '3', '4', '5')))
-
-    def test_customers_setup_zd_26466(self):
-        def make_request(u=None):
-          sleep(10)
-          x = requests.get(testenv["wsgi_server"] + '/')
-          sleep(10)
-          return x.status_code
+        assert len(spans) == 5
+        nums = map(lambda s: s.data["http"]["params"].split("=")[1], spans)
+        assert set(nums) == set(("1", "2", "3", "4", "5"))
+
+    @pytest.mark.skipif(
+        sys.platform == "darwin",
+        reason="Avoiding ConnectionError when calling multi processes of Flask app.",
+    )
```

Comment:
Can we avoid the import of `pytest` module by using `unittest` to skip on condition?

```suggestion
    @unittest.skipIf(
        sys.platform == "darwin",
        reason="Avoiding ConnectionError when calling multi processes of Flask app.",
    )
```

Reply:
🤔 
As the idea is to remove the dependency of the old `unittest` module, I guess it should be interesting to do it another way. I will make the test class no longer dependent on the `unittest` module. 
What do you think @GSVarsha?